File: client/my-family-app/src/app/models/phone-number.ts
/**
 * Represents a phone number, including type, country code, area code, and subscriber number.
 * @author Stella Choi
 */
export class PhoneNumber {
  /** @private @property {string} #phoneType - The type of phone (e.g., mobile, home, work). */
  #phoneType: string;

  /** @private @property {string} #countryCode - The country dialing code. */
  #countryCode: string;

  /** @private @property {string} #areaCode - The area code or region-specific code. */
  #areaCode: string;

  /** @private @property {string} #subscriberNumber - The subscriber's unique phone number. */
  #subscriberNumber: string;

  /**
   * Constructs a new PhoneNumber instance.
   * @param {string} phoneType - The type of phone (e.g., mobile, home, work).
   * @param {string} countryCode - The country dialing code.
   * @param {string} areaCode - The area code or region-specific code.
   * @param {string} subscriberNumber - The subscriber's unique phone number.
   */
  constructor(phoneType: string, countryCode: string, areaCode: string, subscriberNumber: string) {
    this.#phoneType = phoneType;
    this.#countryCode = countryCode;
    this.#areaCode = areaCode;
    this.#subscriberNumber = subscriberNumber;
  }

  /**
   * Gets the type of phone (e.g., mobile, home, work).
   * @returns {string} The phone type.
   */
  public getPhoneType(): string {
    return this.#phoneType;
  }

  /**
   * Gets the type of phone (property accessor).
   * @returns {string} The phone type.
   */
  public get phoneType(): string {
    return this.#phoneType;
  }

  /**
   * Gets the country dialing code.
   * @returns {string} The country code.
   */
  public getCountryCode(): string {
    return this.#countryCode;
  }

  /**
   * Gets the country dialing code (property accessor).
   * @returns {string} The country code.
   */
  public get countryCode(): string {
    return this.#countryCode;
  }

  /**
   * Gets the area code or region-specific code.
   * @returns {string} The area code.
   */
  public getAreaCode(): string {
    return this.#areaCode;
  }

  /**
   * Gets the area code or region-specific code (property accessor).
   * @returns {string} The area code.
   */
  public get areaCode(): string {
    return this.#areaCode;
  }

  /**
   * Gets the subscriber's unique phone number.
   * @returns {string} The subscriber number.
   */
  public getSubscriberNumber(): string {
    return this.#subscriberNumber;
  }

  /**
   * Gets the subscriber's unique phone number (property accessor).
   * @returns {string} The subscriber number.
   */
  public get subscriberNumber(): string {
    return this.#subscriberNumber;
  }

  /**
   * Returns a string representation of the phone number in JSON-like format.
   * @returns {string} A JSON-like string representing the phone number.
   */
  public toString(): string {
    return `"phoneNumber": {
              "type": ${this.#phoneType},
              "country code": ${this.#countryCode},
              "area code": ${this.#areaCode},
              "subscriber number": ${this.#subscriberNumber}
            }`;
  }
}

File: server/my-family-ws/src/main/java/com/bytepushers/family/model/PhoneNumber.java
package com.bytepushers.family.model;

/**
 * Represents a phone number entity in the system.
 * This class stores details about a phone number, including its type, country code, area code,
 * and subscriber number. It extends {@link BaseIdGeneratedValueEntity} to include common entity fields.
 */
public class PhoneNumber extends BaseIdGeneratedValueEntity {

    /** The type of the phone number (e.g., Mobile, Home, Work). */
    private String phoneType;

    /** The country code of the phone number (e.g., +1 for the United States). */
    private String countryCode;

    /** The area code of the phone number. */
    private String areaCode;

    /** The subscriber number of the phone number. */
    private String subscriberNumber;

    /**
     * Constructs a PhoneNumber with the specified details.
     *
     * @param phoneType        the type of the phone number (e.g., Mobile, Home, Work)
     * @param countryCode      the country code of the phone number
     * @param areaCode         the area code of the phone number
     * @param subscriberNumber the subscriber number of the phone number
     */
    public PhoneNumber(String phoneType, String countryCode, String areaCode, String subscriberNumber) {
        this.phoneType = phoneType;
        this.countryCode = countryCode;
        this.areaCode = areaCode;
        this.subscriberNumber = subscriberNumber;
    }

    /**
     * Retrieves the type of the phone number.
     *
     * @return the phone type
     */
    public String getPhoneType() {
        return phoneType;
    }

    /**
     * Sets the type of the phone number.
     *
     * @param phoneType the phone type to set
     */
    public void setPhoneType(String phoneType) {
        this.phoneType = phoneType;
    }

    /**
     * Retrieves the country code of the phone number.
     *
     * @return the country code
     */
    public String getCountryCode() {
        return countryCode;
    }

    /**
     * Sets the country code of the phone number.
     *
     * @param countryCode the country code to set
     */
    public void setCountryCode(String countryCode) {
        this.countryCode = countryCode;
    }

    /**
     * Retrieves the area code of the phone number.
     *
     * @return the area code
     */
    public String getAreaCode() {
        return areaCode;
    }

    /**
     * Sets the area code of the phone number.
     *
     * @param areaCode the area code to set
     */
    public void setAreaCode(String areaCode) {
        this.areaCode = areaCode;
    }

    /**
     * Retrieves the subscriber number of the phone number.
     *
     * @return the subscriber number
     */
    public String getSubscriberNumber() {
        return subscriberNumber;
    }

    /**
     * Sets the subscriber number of the phone number.
     *
     * @param subscriberNumber the subscriber number to set
     */
    public void setSubscriberNumber(String subscriberNumber) {
        this.subscriberNumber = subscriberNumber;
    }

    /**
     * Returns a string representation of this PhoneNumber.
     *
     * @return a string containing the phone number details
     */
    @Override
    public String toString() {
        return "PhoneNumber{" +
                super.toString() +
                ", phoneType='" + phoneType + '\'' +
                ", countryCode='" + countryCode + '\'' +
                ", areaCode='" + areaCode + '\'' +
                ", subscriberNumber='" + subscriberNumber + '\'' +
                '}';
    }
}
